### DISCUSSION TITLE
Battery level indicator question

### DISCUSSION THREAD

**Discussion by marktillo** [ExperienceLovelace/ha-floorplan#204] [Q&A] [answered]
Trying to use the moisture level sensor example and use it to show the charge level of a battery, I’ve copied the code swapped out for my sensor, the element is clickable and shows the battery level, but the box height remains the same can anyone point me in the right direction

**Comment by OzGav** [comment, marked as answer]
When you say you have swapped out your sensor do you mean in the YAML and SVG?

**Reply by marktillo** [reply]
Yes swapped both, hover and click both work on the floorplan but the box height remains the same if you look at the moisture level sensor example the height of the box changes with the value of the moisture sensor

**Reply by marktillo** [reply]
- entity: sensor.moisture_level
        state_action:
          action: call-service
          service: floorplan.style_set
          service_data:
            element: moisture-level-clip-path
            style: |
              >
              var height = Math.ceil(elements['sensor.moisture_level'].getBBox().height);
              return `transform: translate(0, ${height - Math.floor(entity.attributes.level / (100 / height))}px)`;

**Reply by marktillo** [reply]
![E4BC13DB-2970-485D-BB6E-BB88A872A87C](https://user-images.githubusercontent.com/32182400/153867466-62de0f4b-e44e-4f2b-a86a-a4ead6d09d56.png)

**Reply by OzGav** [reply]
You will have to post your yaml and SVG so we can recreate the issue. Also did you change `entity.attributes.level` to what attribute you actually need to reference? Does that attribute provide a value from 0-100?

**Reply by marktillo** [reply]
As you can see I’ve tried with different battery sensors, and slight variations on the code input but with no success

**Reply by OzGav** [reply]
I can't actually see the most important part which is the code in green?

**Reply by OzGav** [reply]
Looking at my battery level sensors I think you might need to change `entity.attributes.level` to `entity.state`

**Reply by marktillo** [reply]
![E0DCF896-93F3-4828-924D-B41AE3532C35](https://user-images.githubusercontent.com/32182400/153868462-4cf583c8-2148-4807-a6c5-496606a405c3.jpeg)

**Reply by marktillo** [reply]
- entity: sensor.iphone_battery_level
                      state_action:
                        action: call-service
                        service: floorplan.style_set
                        service_data:
                          element: iphone-batterylevel-clip-path 
                          style: |
                            >
                            var height = Math.ceil(elements['sensor.iphone_battery_level'].getBBox().height);
                            return `transform: translate(0, ${height - Math.floor(entity.state / (100 / height))}px)`;  
                    - entity: sensor.iphone_battery
                      state_action:
                        action: call-service
                        service: floorplan.style_set
                        service_data:
                          element: sensor.iphone_battery-clip-path 
                          style: |
                            >
                            var height = Math.ceil(elements['sensor.iphone_battery'].getBBox().height);
                            return `transform: translate(0, ${height - Math.floor(entity.state / (100 / height))}px)`;
